spec/services/export_candidate_data_service_spec.rb

```diff
@@ -0,0 +1,175 @@
+require "rails_helper"
+require "zip"
+
+RSpec.describe ExportCandidateDataService do
+  let(:vacancy) { create(:vacancy) }
+  let(:organisation) { vacancy.organisations.first }
+  # referee
+  let(:referee_one) { create(:referee, job_application:, name: "john e. smith") }
+  let(:referee_two) { create(:referee, job_application:, name: "etha may") }
+  let(:referee_no_request) { create(:referee, job_application:) }
+  # requests
+  let(:reference_request_one) do
+    referee_one.create_reference_request!(token: SecureRandom.uuid, status: :received, email: referee_one.email)
+  end
+  let(:reference_request_two) do
+    referee_two.create_reference_request!(token: SecureRandom.uuid, status: :received, email: referee_two.email)
+  end
+  let(:reference_requests) { [reference_request_one, reference_request_two] }
+  # references
+  let(:job_reference_one) { create(:job_reference, :reference_given, referee: referee_one) }
+  let(:job_reference_two) { create(:job_reference, referee: referee_two, can_give_reference: false) }
+  let(:job_references) { [job_reference_one, job_reference_two] }
+  # self disclosure
+  let(:self_disclosure_request) { create(:self_disclosure_request, job_application:) }
+  let(:self_disclosure) { create(:self_disclosure, self_disclosure_request:) }
+  # job application
+  let(:job_application) do
+    create(:job_application, :status_offered, vacancy:, first_name: "John", last_name: "Doe")
+  end
+  let(:job_applications) { [job_application] }
+  # service
+  let(:service) { described_class.new(job_applications) }
+
+  before do
+    referee_no_request
+    reference_requests
+    job_references
+    self_disclosure
+  end
+
+  describe ".call" do
+    it "creates new instance and calls export" do
+      expect(described_class).to receive(:new).with(job_applications).and_call_original
+      expect_any_instance_of(described_class).to receive(:export) # rubocop:disable RSpec/AnyInstance
+      described_class.call(job_applications)
+    end
```

Comment:
Suggestion: Given that this requires the use of `expect_any_instance_of`. Couldn't we replace it with an explicit stub instead? It would still test that a new instance is created and that receives the `export` call.

```ruby
it "creates new instance and calls export" do
  service_instance = instance_double(described_class)
  expect(described_class).to receive(:new).with(job_applications).and_return(service_instance)
  expect(service_instance).to receive(:export)
  described_class.call(job_applications)
end
```